clap_complete/src/dynamic/completer.rs

```diff
@@ -704,3 +723,110 @@ impl CompletionCandidate {
         self.visible
     }
 }
+
+/// User-provided completion candidates for an argument.
+///
+/// This is useful when predefined value hints are not enough.
+pub trait CustomCompleter: core::fmt::Debug + Send + Sync {
```

Comment:
We don't expose the `completer` mod so we need to make `dynamic/mod.rs` re-export these types

Reply:
There is already `pub use completer::*;` in `dynamic/mod.rs`